core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java

```diff
@@ -41,23 +47,37 @@ public ParallelExecutor(ConsensusCommitConfig config) {
     this.parallelExecutorService = parallelExecutorService;
   }
 
-  public void prepare(List<ThrowableRunnable<ExecutionException>> tasks) throws ExecutionException {
-    executeTasks(tasks, config.isParallelPreparationEnabled(), false);
+  public void prepare(List<ParallelExecutorTask> tasks) throws ExecutionException {
+    try {
+      executeTasks(tasks, config.isParallelPreparationEnabled(), false);
+    } catch (CommitConflictException ignored) {
+      // tasks for preparation should not throw CommitConflictException
```

Comment:
We need to ignore `CommitConflictException` in `prepare()`, `commit()`,  and `rollback()`.

Reply:
Thank you!
Ah, now I understand. It's more awkward than I assumed. 😱  
Another option is passing the type (validate, prepare, commit, and rollback) to `executeTask` and letting it decide what to throw based on the given type?
In any case, it doesn't look very nice.

I'm fine with this but I'm also fine with the first one, so you can revert it if you prefer.
Thank you!